### PR TITLE
[ACCSEC-20366] Deploy to MavenCentral

### PR DESCRIPTION
**Pull request by SergioFierro**
## Jira Ticket
- ACCSEC-20366

## Description
This PR adds Maven Central publishing to the `twilio-verify-android` and `twilio-verify-security` project. Previously `twilio-verify-android` and `twilio-verify-security` were published to JCenter, but due to their pending shutdown of the service, we needed to migrate to a new artifact provider.

This PR uses the Gradle Nexus Publish Plugin to manage multitenancy and closing and releasing Next staging repositories.

A few new environment variables were added to the CircleCI settings to assist with this:

- `OSSRH_USERNAME`: The username for the MavenCentral repository. We are actually using a user token here for our twilio-sdk-build user.
- `OSSRH_PASSWORD`: The password for the MavenCentral repository. Again, in this case this is the user token password.
- `SONATYPE_STAGING_PROFILE_ID`: The MavenCentral staging repository profile id. This was obtained from oss.sonatype.org.
- `SIGNING_KEY_ID`: The signing key ID for the twilio-sdk-build user.
- `SIGNING_KEY`: The base64 encoded GPG private key used for signing the artifacts.
- `SIGNING_SECRET_KEY_RING_FILE`: The location where the base64 decoded GPG private key should be written during the and consumed from during the CI process.
- `SIGNING_PASSWORD`: The password for the GPG private key file

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -15,9 +15,10 @@
  */
 
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
-apply(from = "publish.gradle.kts")
 plugins {
   id(Config.Plugins.dokka) version Config.Versions.dokka
+  id(Config.Plugins.nexus) version (Config.Versions.nexus)
+
```
Comment (AuthyApps) [inline build.gradle.kts:21]: <table data-meta="generated_by_Linter">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-android/blob/388e0062c7bc7710990868b5f792c9d176ea58bf/build.gradle.kts#L21"></span>Unexpected blank line(s) before “}”</td>
    </tr>
  </tbody>
</table>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Fix code convention problems</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The Pull Request summary does not contain the commit message convention</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_Linter">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: security/build.gradle.kts [hunk 2]
```diff
@@ -61,33 +62,46 @@ android {
...
+
+      pom.withXml {
+        asNode().apply {
+          appendNode("name", "Twilio Security library")
+          appendNode("description", "Twilio Security library")
+          appendNode("url", "https://github.com/twilio/twilio-verify-android")
+          appendNode("licenses").apply {
+            appendNode("license").apply {
+              appendNode("name", "Apache License, Version 2.0")
+              appendNode("url", "https://github.com/twilio/twilio-verify-android/blob/main/LICENSE")
```
Comment (paynerc) [inline security/build.gradle.kts:84]: Just double checking that Apache 2.0 is the proper license.

File: verify/build.gradle.kts [hunk 3]
```diff
@@ -143,41 +109,63 @@ val sourcesJar by tasks.creating(Jar::class) {
...
+            "Twilio Verify Push SDK helps you verify users by adding a low-friction, secure, " +
+              "cost-effective, \"push verification\" factor into your own mobile application. This fully" +
+              " managed API service allows you to seamlessly verify users in-app via a secure channel," +
+              " without the risks, hassles or costs of One-Time Passcodes (OTPs)."
+          )
+          appendNode("url", "https://github.com/twilio/twilio-verify-android")
+          appendNode("licenses").apply {
+            appendNode("license").apply {
+              appendNode("name", "Apache License, Version 2.0")
+              appendNode("url", "https://github.com/twilio/twilio-verify-android/blob/main/LICENSE")
```
Comment (paynerc) [inline verify/build.gradle.kts:136]: Same double check

**Review by paynerc — APPROVED** [review]
Just double checking that the license is correct. As we have seen, it is very difficult to get the POM files updated if we have any incorrect data in them. Just be extra sure you are happy with the values in the pom before publishing.

Other wise, looks good to me!

**Review by yafuquen — APPROVED** [review]

File: security/build.gradle.kts (hunk 2 above)
Comment (yafuquen) [reply security/build.gradle.kts:84]: Yes! we are open source. Thanks for checking it

**Pull request merged**